### PR TITLE
bpf: propagate src sec id from ingress bpf_overlay to egress bpf_host

### PR DESCRIPTION
**Pull request by jibi**
as in a subsequent change bpf_host will need to access the source identity that was carried over the tunnel

### CODE REVIEW

File: bpf/bpf_overlay.c [hunk 1]
```diff
@@ -465,6 +465,8 @@ int tail_handle_ipv4(struct __ctx_buff *ctx)
 	__s8 ext_err = 0;
 	int ret;
 
+	set_identity_mark(ctx, src_sec_identity, MARK_MAGIC_IDENTITY);
+
```
Comment (julianwiedmann) [inline bpf/bpf_overlay.c]: Won't this conflict with subsequent code that currently reads from the mark, eg. https://github.com/cilium/cilium/blob/bb947ca474a118fcfdb69f14c10c0f4d34d6cc43/bpf/bpf_overlay.c#L333

I think you might be able to push this further down in the function, to where you actually need it (around the call to `ipv4_host_delivery()`) ?

Comment (jibi) [reply bpf/bpf_overlay.c]: for some reason I thought each mark had its own bit, let me move it further down then

**Review by brb — APPROVED** [review]

**Review by aspsk — APPROVED** [review]

File: bpf/bpf_overlay.c [hunk 2]
```diff
@@ -455,6 +455,9 @@ static __always_inline int handle_ipv4(struct __ctx_buff *ctx,
 	/* A packet entering the node from the tunnel and not going to a local
 	 * endpoint has to be going to the local host.
 	 */
+
+	set_identity_mark(ctx, *identity, MARK_MAGIC_IDENTITY);
+
```
Comment (julianwiedmann) [inline bpf/bpf_overlay.c:460]: If this gets scrubbed during the veth traversal, then `to-netdev` would just not see the magic mark and keep the `src_sec_identity` as `UNKNOWN_ID`. I suppose that's alright :).

**Review by julianwiedmann — APPROVED** [review]
Maybe double-check against https://github.com/cilium/cilium/pull/33081, but I think this shouldn't change anything.

File: bpf/bpf_overlay.c (hunk 2 above)
Comment (jibi) [reply bpf/bpf_overlay.c:460]: I think so too

**Pull request merged**